src/test/kotlin/dev/supergrecko/kllvm/ir/BuilderTest.kt

```diff
@@ -33,11 +38,16 @@ class BuilderTest {
         // that holds the reference is different
         // TODO?: Implement equals/hashCode for Builder by comparing underlying
         //   refs?
-        assertEquals(builder.getInsertBlock()?.ref, basicBlock.ref)
+        val ref = when (val block = builder.getInsertBlock()) {
+            is Some<BasicBlock> -> block.t.ref
+            is None -> fail()
+        }
+
+        assertEquals(ref, basicBlock.ref)
```

Comment:
This is what the API usage will look like now. Instead of possibly ignoring a null at runtime we deal with it as soon as we receive it.

Using this paradigm we also prefer using when blocks for error handling over exceptions. This could be both a benefit or a drawback.